### PR TITLE
libsbol v2.3.0.0

### PR DESCRIPTION
**Pull request by bbartley**
- [x ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/libsbol.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+      using namespace sbol;
+
+      int main() {
+        Document& doc = *new Document();
+        doc.write("test.xml");
+        return 0;
+      }
+    EOS
+    if ENV.compiler == :gcc
```
Comment (fxcoudert) [inline Formula/libsbol.rb]: The case for `:gcc` (with the extra `"-Wl,-rpath=#{HOMEBREW_PREFIX}/lib,-rpath=#{lib}"`) should not be necessary. The compiler should know where it find its own runtime libraries.

File: Formula/libsbol.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      system ENV.cxx, "test.cpp", "-o", "test", "-std=c++11", "-I/System/Library/Frameworks/Python.framework/Headers", "-I#{HOMEBREW_PREFIX}/include/curl", "-I#{HOMEBREW_PREFIX}/include", \
+      "-I#{HOMEBREW_PREFIX}/include/raptor2", "-I#{include}/sbol", "-L#{HOMEBREW_PREFIX}/lib", "-L#{lib}", "-Wl,-rpath=#{HOMEBREW_PREFIX}/lib,-rpath=#{lib}", "-ljsoncpp", "-lcurl", \
+      "-lraptor2", "-lsbol"
+    else
+      system ENV.cc, "test.cpp", "-o", "test", "-std=c++11", "-I/System/Library/Frameworks/Python.framework/Headers", "-I#{HOMEBREW_PREFIX}/include/curl", "-I#{HOMEBREW_PREFIX}/include", \
```
Comment (fxcoudert) [inline Formula/libsbol.rb]: - `ENV.cxx`
- Remove `-lc++`
- Why does it do `-I#{HOMEBREW_PREFIX}/include/curl` when it doesn't depend on Homebrew curl? Likely remove
- `-I#{Formula["raptor"].opt_include}/raptor2`
- Remove `-I#{HOMEBREW_PREFIX}/include` and include specific packages' include (like raptor)
- `-I#{include}/sbol` should probably be `-I#{include}`, and the test adapted to include `<sbol/sbol.h>`?
- `-L#{HOMEBREW_PREFIX}/lib`: nope

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by bbartley** [comment]
Hi FX,

Thanks for the comments.  I implemented your recommendations except for:

- Remove -I#{HOMEBREW_PREFIX}/include and include specific packages' include (like raptor)
- -I#{Formula["raptor"].opt_include}/raptor2

The pattern -I#{Formula["raptor"].opt_include}/raptor2 results in "error: 'raptor2.h' file not found with <angled> include; use "quotes" instead".  (The source code uses #include <raptor2.h>) The pattern -I#{HOMEBREW_PREFIX}/include/raptor2 compiles with no problem.

Thanks!
Bryan

**Comment by fxcoudert** [comment]
There is no reason that `-I#{Formula["raptor"].opt_include}/raptor2` shouldn't work, and it does indeed work for me locally. Can you re-test? It may be a problem with your local setup.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @bbartley for the pull request!

**Pull request closed without merging**